source/validate_image.cpp

```diff
@@ -980,6 +1055,13 @@ spv_result_t ImagePass(ValidationState_t& _,
             << "Corrupt image type definition";
       }
 
+      if (info.dim == SpvDimSubpassData) {
```

Comment:
I reviewed other places which should check for SubpassData.   
An image with Dim=SubpassData is required to have Sampled = 2 (i.e. unsampled), so that already excludes the majority of image operations.   OpImageWrite already bans an image with Dim = SubpassData.  

The only interesting case I found is OpImageSparseRead;  I filed https://github.com/KhronosGroup/SPIRV-Tools/issues/1022 to ban the use of images with Dim=SubpassData.  That's not explicitly spec'd anywhere but I'm pretty sure the (Vulkan) use case for SubpassData is different from sparse memory.